doc/source/dev/contributor/quickstart_mac.rst

```diff
@@ -0,0 +1,109 @@
+.. _quickstart-mac:
+
+================================================
+Development Environment Quickstart Quide (macOS)
+================================================
```

Comment:
New. Linked only from CTOC right now, but basically is a more detailed version of the macOS instructions (`doc/source/building/macosx.rst`) linked from `doc/source/building/index.rst`. For now, I'm leaving these both in the documentation. CTOC leads new contributors to this "Quickstart Guide" if they have macOS and to the other build docs otherwise. I will consider replacing `doc/source/building/macosx.rst` with this someday, but probably only after I can also replace `doc/source/building/linux.rst`.